### PR TITLE
Change interface to support rule metas

### PR DESCRIPTION
**Pull request by kmcq**
## Why?

Currently `Yara.test` returns an array of the rule names that matched the given string. This does not allow us to know a rule's metas or strings values. This PR changes this method to return an array of `Yara::ScanResult` objects, which have methods to know the rule's name, if it matched, and any metas.

## How?

- Return a Yara::ScanResult object instead of only rule name
- Return rule metas as a hash of name => value

### CODE REVIEW

**Comment by kmcq** [comment]
@jonmagic I don't have permissions to request a review from you :)

File: lib/yara/scan_result.rb
Comment (jonmagic) [inline lib/yara/scan_result.rb:5]: I kinda wish we didn't create a ScanResult for the SCAN_FINISHED callback_type. What do you think about only constructing one if RULE_MATCHING or RULE_NOT_MATCHING?

**Review by jonmagic — APPROVED** [review]
Approving but think we should update to only create a ScanResult for RULE_MATCHING and RULE_NOT_MATCHING message types.

File: lib/yara/scan_result.rb
Comment (kmcq) [reply lib/yara/scan_result.rb]: I like that 👍 Updated in https://github.com/jonmagic/yara-ffi/pull/4/commits/90cd1acacfcab1f268012c1f6b71d1fa7d25b860

**Comment by kmcq** [comment]
See https://github.com/jonmagic/yara-ffi/pull/5#issuecomment-904653514 regarding the failing builds. Our builds are messed up, but let's not let that block this PR. @jonmagic this is ready to be merged and pushed to rubygems 🚀

**Pull request merged**
